### PR TITLE
feat: add Playbooks/Enrich-AzureResourceGraph

### PR DESCRIPTION
**Pull request by juju4**
Change(s):
   - add Playbooks/Enrich-AzureResourceGraph

   Reason for Change(s):
This LogicApp is querying Azure ResourceGraph and return typical azure information related to the resource like subscription, resourcegroup, tags and management groups.
It is encapsulated in other Logic app to enrich Sentinel incident (like Enrich-AzureResourceGraph-Incident).

   Testing Completed:
   - Yes. used in production for many months

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes. Deployed in production with Sentinel Repository feature, precommit and psrule

### CODE REVIEW

File: Playbooks/Enrich-AzureResourceGraph/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,420 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
+        "title": "Query Azure Resource Graph with HTTP input and output",
+        "description": "This playbook will query Azure Resource Graph with HTTP input and output.",
```
Comment (rahul0216) [inline Playbooks/Enrich-AzureResourceGraph/azuredeploy.json:6]: From the description, I'm not sure what this playbook is doing. Can you make it more comprehensible?

**Review by rahul0216 — COMMENTED** [review]
Keyvault secret name is fixed, but nowhere it is motioned to create those secrets with those fixed name. You can take the secret name as parameter, as well or at least capture those steps in readme file.

**Comment by v-rbajaj** [comment]
Hi @juju4, can you please address @rahul0216's comment?

**Comment by v-rbajaj** [comment]
Hi @juju4, can you please address @rahul0216's comment?

**Comment by juju4** [comment]
ARM template prerequisites has
> "1. Set service principal with Reader role to query resourcegraph.\n2. Set keyvault to store client id and secret.\n3. Pass those parameters at deployment time."

Readme had
> * Service principal client id and secret stored in Azure keyvault (Possible change to Managed Identity as supported by HTTP block)

Slightly edited the later to add expected keyvault secret names.

Please withold on merge as I'm reviewing playbook to have resourcegraph as a an optional playbook parameter.

File: Playbooks/Enrich-AzureResourceGraph/azuredeploy.json (hunk 1 above)
Comment (juju4) [reply Playbooks/Enrich-AzureResourceGraph/azuredeploy.json:6]: I'm unclear how to reword differently.
This is to make query to Azure Resource Graph - https://learn.microsoft.com/en-us/azure/governance/resource-graph/overview

**Comment by rahul0216** [comment]
@juju4 I'll wait for your confirmation before approving this PR, as you may have some changes.

**Comment by v-rbajaj** [comment]
Hi @juju4, waiting for confirmation from your side.

**Comment by v-rbajaj** [comment]
Hi @juju4, can you please provide confirmation?

**Comment by juju4** [comment]
Please give me one or two weeks to work on testing.

**Comment by v-atulyadav** [comment]
Hi @juju4, thanks for acknowledging.

**Comment by juju4** [comment]
Code and docs update but as per support delay, I won't be able to test until an extra week.
So probably 2 weeks more before giving go.

**Comment by v-rbajaj** [comment]
@juju4, Thanks for updating us.

**Comment by v-rbajaj** [comment]
Hi @juju4, is there any update on this PR?

**Comment by juju4** [comment]
updated pushed.
Thanks for your patience.

File: Playbooks/Enrich-AzureResourceGraph/azuredeploy.json (hunk 1 above)
Comment (rahul0216) [reply Playbooks/Enrich-AzureResourceGraph/azuredeploy.json:6]: Something like this,
"This playbook queries Azure Resource Graph and returns azure information related to the resource like Subscription, Resourcegroups, Tags and Management groups."

File: Playbooks/Enrich-AzureResourceGraph/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,630 @@
...
+                                ]
+                            },
+                            "type": "Http",
+                            "inputs": {
+                                "body": "resource=https://management.azure.com&grant_type=client_credentials&client_id=@{body('Get_secret_azureresourcegraph-clientid')?['value']}&client_secret=@{body('Get_secret_azureresourcegraph-clientsecret')?['value']}",
+                                "headers": {
+                                    "Content-Type": "application/x-www-form-urlencoded"
+                                },
+                                "method": "POST",
+                                "uri": "https://login.microsoftonline.com/29208c38-8fc5-4a03-89e2-9b6e8e4b388b/oauth2/token"
```
Comment (rahul0216) [inline Playbooks/Enrich-AzureResourceGraph/azuredeploy.json:319]: Tenant "29208c38-8fc5-4a03-89e2-9b6e8e4b388b" is hard coded. Will it work for other tenants?

File: Playbooks/Enrich-AzureResourceGraph/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,630 @@
...
+                            "runAfter": {
+                                "Parse_JSON": [
+                                    "Succeeded"
+                                ]
+                            },
+                            "type": "Http",
+                            "inputs": {
+                                "body": {
+                                    "managementGroups": [
+                                        "29208c38-8fc5-4a03-89e2-9b6e8e4b388b"
```
Comment (rahul0216) [inline Playbooks/Enrich-AzureResourceGraph/azuredeploy.json:332]: Similarly hard coded value '29208c38-8fc5-4a03-89e2-9b6e8e4b388b'.

**Review by rahul0216 — COMMENTED** [review]
Please address the inline comment.

**Comment by v-atulyadav** [comment]
Hi @juju4, please act on @rahul0216's comments. Thanks

File: Playbooks/Enrich-AzureResourceGraph/azuredeploy.json (hunk 2 above)
Comment (juju4) [reply Playbooks/Enrich-AzureResourceGraph/azuredeploy.json]: fixed with Playbooks/Notify-ASCAlertAzureResource/azuredeploy.json as model

**Comment by rahul0216** [comment]
Playbook deployment is failing with following error,
image

**Comment by juju4** [comment]
Fixed but that means Playbooks/Notify-ASCAlertAzureResource/azuredeploy.json is likely broken as it was following same parameter model.

**Comment by juju4** [comment]
Last update deploys fine but is not functional. Tenant parameter not replaced in url path.
I wonder if there is a recommended way to handle that or a working model template unlike Playbooks/Notify-ASCAlertAzureResource

**Comment by rahul0216** [comment]
@juju4 parameter values are replaced during execution. Did you run the playbook and test after update? 
You can refer this playbook, https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Australian%20Cyber%20Security%20Centre/Playbooks/AusCtisExportTaggedIndicators/azuredeploy.json

It is not exact scenario but give you an idea how parameters are used in logic app. 
Also, Please attach a screenshot of successful run once you have one.

**Comment by v-rbajaj** [comment]
Hi @juju4, please look into Rahul's comment, please work on the requested changes.

**Comment by v-rbajaj** [comment]
Hi @juju4, please look into Rahul's comment and act on it accordingly.

**Comment by v-rbajaj** [comment]
Hi @juju4, waiting for some update from you on this PR

**Comment by v-rbajaj** [comment]
Hi @juju4, please act on this PR.

**Comment by juju4** [comment]
fix for tenantid done. tested in deployment pipeline and run.
should be good for merge.

**Comment by rahul0216** [comment]
@juju4 Thanks for the updates. These looks good. Can you please share screenshot of a successful run?

**Comment by juju4** [comment]
Here are some screenshots
Screenshot run1
Screenshot run2

**Review by rahul0216 — APPROVED** [review]
@juju4 Thanks for the screenshot.

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**